test/functional/ui/cmdline_spec.lua

```diff
@@ -157,6 +157,27 @@ describe('external cmdline', function()
     end)
   end)
 
+  it("redraws statusline on entering", function()
+    command('set laststatus=2')
+    command('set statusline=%{mode()}')
+    feed(':')
+    screen:expect([[
+                               |
+      {1:~                        }|
+      {1:~                        }|
+      {3:c^                        }|
```

Comment:
This cursor position is a bit annoying, but maybe leave it as is for now, as #8221 will change cursor management in general and with ext_cmdline specifically anyway.